app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt

```diff
@@ -36,6 +36,15 @@ class SmokeTest {
     private var awesomeBar: ViewVisibilityIdlingResource? = null
     private var searchSuggestionsIdlingResource: RecyclerViewIdlingResource? = null
 
+    // This finds the dialog fragment child of the homeFragment, otherwise the awesomeBar would return null
+    private fun getAwesomebarView(): View? {
+        val homeFragment = activityTestRule.activity.supportFragmentManager.primaryNavigationFragment
+        val searchDialogFragment = homeFragment?.childFragmentManager?.fragments?.first {
+            it.javaClass.simpleName == "SearchDialogFragment"
+        }
+        return searchDialogFragment?.view?.findViewById(R.id.awesome_bar)
+    }
+

```

Comment:
Fragments still relatively new to me, so I'm not sure if there's any possible problems with this – but I don't think there's standing out here that looks off.

Reply:
Actually, that is a part from a different PR that has already landed, here I just moved it. Sorry for the confusion.